ogre2/src/Ogre2RenderTarget.cc

```diff
@@ -110,12 +110,144 @@ Ogre2RenderTarget::~Ogre2RenderTarget()
 //////////////////////////////////////////////////
 void Ogre2RenderTarget::BuildCompositor()
 {
-  this->UpdateShadowNode();
-
   auto engine = Ogre2RenderEngine::Instance();
   auto ogreRoot = engine->OgreRoot();
   Ogre::CompositorManager2 *ogreCompMgr = ogreRoot->getCompositorManager2();
 
+  this->UpdateBackgroundMaterial();
+
+  bool validBackground = this->backgroundMaterial &&
+      !this->backgroundMaterial->EnvironmentMap().empty();
+
+  // The function build a similar compositor as the one defined in
+  // ogre2/media/2.0/scripts/Compositors/PbsMaterials.compositor
+  // but supports an extra quad pass to render the skybox cubemap if
+  // sky is enabled.
+  // todo(anyone) Note the definition programmatically created here
+  // replaces the one defined in the script so it maybe safe to remove the
+  // PbsMaterials.compositor file
+  std::string wsDefName = "PbsMaterialWorkspace_" + this->Name();
+  this->ogreCompositorWorkspaceDefName = wsDefName;
+  if (!ogreCompMgr->hasWorkspaceDefinition(wsDefName))
+  {
+    // PbsMaterialsRenderingNode
+    std::string nodeDefName = wsDefName + "/PbsMaterialsRenderingNode";
```

Comment:
There are a few strings being reused here too, we should consider storing them in a common place.

Reply:
done in 0823885 and fbae460